### ISSUE TITLE
false positive `reportIncompatibleUnannotatedOverride` with split definition in the base class

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#1510]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&reportUnannotatedClassAttribute=false&code=MYGwhgzhAECCBcAoaLpmgXmgRkc1AJgKYBm0JAFBESCQJRKpPTW0B06WARGF3qJBgAhCrAb4UnaDz5A)

```python
class A:
    a = 1

    def f(self):
        self.a = "a"

class B(A):
    a = "a" # reportIncompatibleUnannotatedOverride, expect no error
```

**Comment by DetachHead** [comment]
unlike the example in #1509, `B.a`'s type doesn't get union'd with the type from the other assignment

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&reportUnannotatedClassAttribute=false&code=GYJw9gtgBALgngBwJYDsDmUkQWEMogCmAboQIYA2A%2BvAoQFCMDGFZAzm1AIIBc9UAqGSgBeKAEZGgqABNCwKMAAUbQhWABKPtOmr1AOmFiARGWPNWHKACElXLf0FGop84yKlKNRISW2NhhpQAMRQbDAgQA)

```python
from typing import reveal_type


class A:
    a = 1

    def f(self):
        self.a = "a"

class B(A):
    a = "a"


reveal_type(B().a) # str
```


so i think the current behavior is correct

**Comment by KotlinIsland** [comment]
the current behavior is not optimal though. `B.a` should infer to be the same as the super type
